tests/unit/classes/view/members_test.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+require_once($CFG->dirroot . '/mod/turnitintooltwo/classes/view/members.php');
+require_once($CFG->dirroot . '/mod/turnitintooltwo/turnitintooltwo_assignment.class.php');
+require_once($CFG->dirroot . '/mod/turnitintooltwo/turnitintooltwo_view.class.php');
+require_once($CFG->dirroot . '/webservice/tests/helpers.php');
+require_once($CFG->dirroot . '/mod/lti/lib.php');
+
+/**
+ * Tests for classes/view/members
+ *
+ * @package turnitintooltwo
+ */
+class mod_turnitintooltwo_view_members_testcase extends externallib_advanced_testcase {
+    /**
+     * Test display role given returns as the expected Turnitin role
+     */
+    public function test_get_role_for_display_role() {
+        $members = new members_view();
+
+        $role = $members->get_role_for_display_role(null);
+        $this->assertEquals('Learner', $role);
+
+        $role = $members->get_role_for_display_role("tutors");
+        $this->assertEquals('Instructor', $role);
+
+        $role = $members->get_role_for_display_role("students");
+        $this->assertEquals('Learner', $role);
+    }
+
+    /**
+     * Test given a role the correct intro message for the members view is
+     * generated.
+     */
+    public function test_build_intro_message() {
+        $members = new members_view();
+
+        $message = $members->build_intro_message();
+        $this->assertEquals('<div id="general" class="box generalbox boxaligncenter">The selected Users below are enrolled on this Turnitin Class. Enrolled students can gain access to this class by logging in to the Turnitin web site.</div>', $message);
```

Comment:
Better to use the get_string() function for the text in case we change the English strings.